### PR TITLE
GEODE-5018: Fixing PIMPL pattern for Cache

### PR DESCRIPTION
**Pull request by mcmellawatt**

### CODE REVIEW

File: cppcache/include/geode/Cache.hpp [hunk 1]
```diff
@@ -76,7 +78,8 @@ class APACHE_GEODE_EXPORT Cache : public GeodeCache {
    * @param regionShortcut
    *        To create the region specific type, @see RegionShortcut
    */
-  virtual RegionFactory createRegionFactory(RegionShortcut regionShortcut);
```
Comment (pivotal-jbarrett) [inline cppcache/include/geode/Cache.hpp]: Region shortcut is an enum type. The size of the ref is actually more expensive than the enum value.

File: cppcache/integration-test/testExpiration.cpp [hunk 2]
```diff
@@ -378,8 +395,9 @@ BEGIN_TEST(TEST_EXPIRATION)
     //////////////
     RegionAttributes attrs_18;
     // ettl = 0, eit = 0, rttl = 0, reit = 0
-    attrs_18 = setRegionAttributesTimeouts(std::chrono::seconds(6), std::chrono::seconds(3),
```
Comment (pivotal-jbarrett) [inline cppcache/integration-test/testExpiration.cpp]: Seems like a lot of formatting changes. Are you using clang-format?

Comment (mcmellawatt) [reply cppcache/integration-test/testExpiration.cpp]: Yes I ran all the source I touched through clang-format.

File: cppcache/include/geode/Cache.hpp (hunk 1 above)
Comment (pivotal-jbarrett) [reply cppcache/include/geode/Cache.hpp]: That’s fine, it’s a small enum. Though I would expect a forward to work. See docs for `enum class`.

File: cppcache/integration-test/testExpiration.cpp (hunk 2 above)
Comment (pivotal-jbarrett) [reply cppcache/integration-test/testExpiration.cpp]: What version of clang-format?

File: cppcache/include/geode/Cache.hpp (hunk 1 above)
Comment (mcmellawatt) [reply cppcache/include/geode/Cache.hpp]: You are right - forward declaration does work, CLion was giving me a false error, but it compiles fine.  I'll watch out for that sort of thing in the future.

**Pull request merged**